src/sbvr-api/permissions.ts

```diff
@@ -1089,15 +1091,17 @@ const getReqPermissions = (req: PermissionReq, odataBinds: ODataBinds = []) =>
 	Promise.join(
 		getGuestPermissions(),
 		Promise.try(() => {
-			if (
-				req.apiKey != null &&
-				req.apiKey.permissions != null &&
-				req.apiKey.permissions.length > 0
-			) {
-				return getApiKeyActorId(req.apiKey.key);
+			// TODO: Remove this extra actor ID lookup making actor non-optional and updating open-balena-api.
+			if (req.apiKey == null) {
+				return null;
+			}
+			if (req.apiKey.actor == null) {
```

Comment:
I'd join this with the other check as
```suggestion
			if (req.apiKey != null && req.apiKey.actor == null) {
```

Reply:
Addressed.